### ISSUE TITLE
ParFlow make issue using latest commits

### ISSUE THREAD

**Issue by xxirii** [parflow/parflow#225]
Dear ParFlow developers,

I am a recent user of ParFlow. I was using commit [9c0b0f54acd6fbef2902c3d52a6bc82c285b335b](9c0b0f54acd6fbef2902c3d52a6bc82c285b335b) of Master branch until recently. A recent pull to the lastest commit is preventing me to finish the make process.

The compilation (and linking) issue occurs in the test8 phase :

Scanning dependencies of target test8
[ 10%] Building C object pfsimulator/amps/test/src/CMakeFiles/test8.dir/test8.c.o
[ 10%] Linking C executable test8

Find attached the log from terminal.

[make_log.txt](https://github.com/parflow/parflow/files/4313308/make_log.txt)

I compile with the following cmake configuration:

```
cmake ../ -DPARFLOW_AMPS_LAYER=mpi1 -DPARFLOW_HAVE_CLM=ON -DCMAKE_INSTALL_PREFIX=$PARFLOW_DIR -DHDF5_ROOT=$HDF5_HOME -DSILO_ROOT=$SILO_HOME -DPARFLOW_ENABLE_NETCDF=ON -DNETCDF_DIR=$NETCDF_HOME -DHYPRE_ROOT=$HYPRE_HOME
```

For both cases, I use the same environment:
- GCC 7
- MPICH 3.3

I can work with the previous commit but I think I should keep up-to-make at some point.

Thank you in advance

**Comment by smithsg84** [comment]
Sorry about bug, something appears to have changed CMake MPI detection but I'm not sure what yet.  It is not compiling with MPI libraries resulting in the errors.  A possible work-around to try is manually setting compilers CMake is using to be the MPI wrappers:

FC=mpif90 CC=mpicc CXX=mpiCC cmake ......

**Comment by xxirii** [comment]
Indeed it was a problem due to the wrappers. I did not suspect this at first since the cmake summary was ok for me. Thank you your help.

**Comment by smithsg84** [comment]
This was caused by the addition of tests which were not including MPI libraries in link.

**Comment by smithsg84** [comment]
Fixed in #227
